qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -767,3 +773,130 @@ def ds_with_vals(self, dataset):
     def test_get_data_with_start_and_end_args(self, ds_with_vals,
                                               start, end, expected):
         assert expected == ds_with_vals.get_data(self.x, start=start, end=end)
+
+
```

Comment:
Why do the tests have not have assertions on the actual data being retrieved (and it's type)? Is it because we just trust `get_data`? also no tests cover the start/end feature.